### PR TITLE
Fixes patching LttP on a fresh AP install

### PR DESCRIPTION
**Pull request by zig-for**
## What is this fixing or adding?
```python .\SNIClient.py .\output\AP_07758176404715800194_P1_YourName1.aplttp

Archipelago (0.4.2) logging initialized on Windows-10-10.0.19045-SP0 running Python 3.8.10
Patch file was supplied. Creating sfc rom..
Wrote rom file to .\output\AP_07758176404715800194_P1_YourName1.sfc
Uncaught exception
Traceback (most recent call last):
  File ".\SNIClient.py", line 726, in <module>
    asyncio.run(main())
  File "D:\Python38\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "D:\Python38\lib\asyncio\base_events.py", line 616, in run_until_complete
    return future.result()
  File ".\SNIClient.py", line 698, in main
    adjustedromfile, adjusted = get_alttp_settings(romfile)
  File "D:\git\Archipelago\worlds\alttp\Client.py", line 593, in get_alttp_settings
    setattr(last_settings, option_name, getattr(base_settings, option_name))
AttributeError: 'dict' object has no attribute 'heartcolor'
```

Cleans up usage of lttp adjuster settings and hardcoded values that could be different than the manual calls to the adjuster. Also implements --disablemusic properly

## How was this tested?
Genned. After the first patch, settings appeared to be kosher. Mucked around with the --(disable)music option a bit

## If this makes graphical changes, please attach screenshots.
```
usage: LttPAdjuster.py [-h] [--baserom BASEROM] [--loglevel [{error,info,warning,debug}]] [--menuspeed [{normal,instant,double,triple,quadruple,half}]] [--quickswap] [--deathlink] [--allowcollect]
                       [--music | --disablemusic] [--triforcehud [{normal,hide_goal,hide_required,hide_both}]] [--enableflashing] [--heartbeep [{double,normal,half,quarter,off}]]
                       [--heartcolor [{red,blue,green,yellow,random}]] [--ow_palettes {default,random,blackout,puke,classic,grayscale,negative,dizzy,sick}]
                       [--shield_palettes {default,random,blackout,puke,classic,grayscale,negative,dizzy,sick}] [--sword_palettes {default,random,blackout,puke,classic,grayscale,negative,dizzy,sick}]
                       [--hud_palettes {default,random,blackout,puke,classic,grayscale,negative,dizzy,sick}] [--uw_palettes {default,random,blackout,puke,classic,grayscale,negative,dizzy,sick}] [--sprite SPRITE]
                       [--oof OOF] [--names NAMES] [--update_sprites]
                       [rom]

positional arguments:
  rom                   Path to an ALttP rom to adjust.
~snip~
  --music, --disablemusic
                        Enables/Disables game music. (default: True)
~snip~
```

### CODE REVIEW

**Comment by zig-for** [comment]
I can improve the behavior. Could I get a description for what is supposed to happen? If you’ve never used the adjuster, don’t save settings and don’t show the UI? Or is the new behavior intended/desired?Sent from my iPhoneOn Aug 22, 2023, at 11:45 AM, Fabian Dill ***@***.***> wrote:﻿
Yeah, when getting the adjuster settings it now populates it with the default, so it checking for an empty dict will now always trigger, as the dict is populated and bring up the UI.

**Comment by Berserker66** [comment]
Restoring the old behaviour would be appropriate - if no custom adjuster settings exist, skip the UI

**Comment by zig-for** [comment]
Ok, I think I fixed things up now:
 * Running LttpAdjuster`.py` will use the persisted adjustersettings as defaults (this _might_ be new behavior, but is probably wanted)
 * Added `sprite_pool` and `auto_apply` to the arguments, so that they get correctly populated with defaults
 * Running the client on a fresh install will ignore the adjuster as no settings are stored
 * Running the client with adjuster settings will overlay those settings on top of the defaults

**Comment by zig-for** [comment]
Possible further improvements:
* Warning/removal of arguments that don't exist

**Comment by zig-for** [comment]
Someone who's a better writer than me should come up with nicer help text. :)

File: worlds/alttp/Client.py [hunk 1]
```diff
@@ -581,31 +581,25 @@ async def game_watcher(self, ctx):
...
-    if last_settings:
+    if vars(Utils.get_adjuster_settings_no_defaults(GAME_ALTTP)):
+        last_settings = Utils.get_adjuster_settings(GAME_ALTTP)
+
+        allow_list = {"music", "menuspeed", "heartbeep", "heartcolor", "ow_palettes", "quickswap",
+                    "uw_palettes", "sprite", "sword_palettes", "shield_palettes", "hud_palettes",
+                    "reduceflashing", "deathlink", "allowcollect", "oof"}
         choice = 'no'
-        if not hasattr(last_settings, 'auto_apply') or 'ask' in last_settings.auto_apply:
+        if 'ask' in last_settings.auto_apply:
```
Comment (zig-for) [inline worlds/alttp/Client.py:592]: Giving some side-eye to all this `if x in y`

File: worlds/alttp/Client.py [hunk 2]
```diff
@@ -685,17 +679,17 @@ def onButtonClick(answer: str = 'no'):
...
 
         if 'yes' in choice:
+            import LttPAdjuster
             from worlds.alttp.Rom import get_base_rom_path
             last_settings.rom = romfile
             last_settings.baserom = get_base_rom_path()
             last_settings.world = None
 
-            if hasattr(last_settings, "sprite_pool"):
+            if last_settings.sprite_pool:
```
Comment (zig-for) [inline worlds/alttp/Client.py:688]: I think there was maybe a bug here if you had an empty sprite_pool before.

File: worlds/alttp/Client.py [hunk 3]
```diff
@@ -581,31 +581,25 @@ async def game_watcher(self, ctx):
...
-                sprite_pool = {}
-                for sprite in last_settings.sprite_pool:
-                    if sprite in sprite_pool:
-                        sprite_pool[sprite] += 1
-                    else:
-                        sprite_pool[sprite] = 1
-                    if sprite_pool:
-                        printed_options["sprite_pool"] = sprite_pool
+            
+            sprite_pool = {}
```
Comment (zig-for) [inline worlds/alttp/Client.py:595]: Ditto for removing this special case. Just show the pool as it exists in the yaml

File: LttPAdjuster.py [hunk 4]
```diff
@@ -560,33 +602,8 @@ def RomSelect():
 
     return romFrame, romVar
 
-
 def get_rom_options_frame(parent=None):
     adjuster_settings = get_adjuster_settings(GAME_ALTTP)
-    defaults = {
```
Comment (zig-for) [inline LttPAdjuster.py:566]: Hooray, no more duplicated default args.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**